--- pcp-3.9.6/configure.ac.orig	2014-06-19 19:02:57.625789139 +0200
+++ pcp-3.9.6/configure.ac	2014-06-19 19:06:56.395784163 +0200
@@ -937,17 +937,7 @@
 NR == 1			{ if ($1 != "UID" && $1 != "USER") exit
 			  if ($2 != "PID") exit
 			}
-# Unix variants
-$2 == 1 && / init/	{ print "OK"; exit }
-$2 == 1 && / \/etc\/init/ { print "OK"; exit }
-# Fedora 9
-$2 == 1 && / \/sbin\/init/ { print "OK"; exit }
-# Fedora 15
-$2 == 1 && / \/bin\/systemd/ { print "OK"; exit }
-# Fedora 17
-$2 == 1 && / \/usr\/lib\/systemd\/systemd/ { print "OK"; exit }
-# Debian sid
-$2 == 1 && / \/lib\/systemd\/systemd/ { print "OK"; exit }'`
+$2 ~ /^[[0-9]]+$/ && /configure/	{ print "OK"; exit }'`
 	if test "$ans" = OK
 	then
 	    pcp_ps_have_bsd=false
@@ -962,10 +952,7 @@
 NR == 1			{ if ($1 != "UID" && $1 != "USER") exit
 			  if ($2 != "PID") exit
 			}
-$2 == 1 && / init/	{ print "OK"; exit }
-$2 == 1 && / \/sbin\/init/ { print "OK"; exit }
-$2 == 1 && / \/sbin\/launchd/ { print "OK"; exit }
-$2 == 1 && / \/init/	{ print "OK"; exit }'`
+$2 ~ /^[[0-9]]+$/ && /configure/	{ print "OK"; exit }'`
 	if test "$ans" = OK
 	then
 	    pcp_ps_have_bsd=true
