--- pcp-3.9.6/configure.ac.orig	2013-09-15 16:11:00.907263278 +0200
+++ pcp-3.9.6/configure.ac	2013-09-15 16:12:28.537259600 +0200
@@ -127,7 +127,7 @@
     test -f /etc/slackware-version && target_distro=slackware
     test -f /etc/gentoo-release && target_distro=gentoo
     test -f /etc/mandriva-release && target_distro=mandriva
-    export CFLAGS="-fPIC -fno-strict-aliasing -D_GNU_SOURCE"
+    export CFLAGS="$CFLAGS -fPIC -fno-strict-aliasing -D_GNU_SOURCE"
 elif test $target_os = darwin
 then
     AC_DEFINE(IS_DARWIN)
