--- pcp-4.1.0/configure.ac~	2018-06-15 03:48:02.000000000 +0200
+++ pcp-4.1.0/configure.ac	2018-07-04 17:50:38.620258455 +0200
@@ -278,7 +278,7 @@ then
     test -f /etc/gentoo-release && target_distro=gentoo
     test -f /etc/mandriva-release && target_distro=mandriva
     test -f /etc/arch-release && target_distro=arch
-    export CFLAGS="-fPIC -fno-strict-aliasing -D_GNU_SOURCE"
+    export CFLAGS="$CFLAGS -fPIC -fno-strict-aliasing -D_GNU_SOURCE"
     pcp_platform_paths='/usr/bin/X11:/usr/local/bin'
     pcp_ps_all_flags=-efw
 elif test $target_os = darwin
