--- pcp-7.0.1/src/include/builddefs.in.orig	2025-09-21 07:17:17.000000000 +0200
+++ pcp-7.0.1/src/include/builddefs.in	2025-10-03 18:50:35.362082340 +0200
@@ -713,13 +713,7 @@ SETUP_PY_BUILD_OPTIONS += --include-dirs
 SETUP_PY_BUILD_OPTIONS += --library-dirs=$(TOPDIR)/src/libpcp/src:$(TOPDIR)/src/libpcp_pmda/src:$(TOPDIR)/src/libpcp_gui/src:$(TOPDIR)/src/libpcp_import/src:$(TOPDIR)/src/libpcp_mmv/src
 SETUP_PY_INSTALL_OPTIONS = --skip-build
 SETUP_PY_INSTALL_OPTIONS += --root="$${DIST_ROOT:-/}"
-ifeq "$(PYTHON_PREFIX)" "/usr"
-ifeq "$(PACKAGE_DISTRIBUTION)" "debian"
-SETUP_PY_INSTALL_OPTIONS += --install-layout=deb
-endif
-else
 SETUP_PY_INSTALL_OPTIONS += --prefix=$(PYTHON_PREFIX)
-endif
 # RPM byte-compiles and installs results in our DIST_ROOT, cater for this:
 ifeq "$(shell [ '$(TARGET_OS)' = linux -a '$(PACKAGE_DISTRIBUTION)' != gentoo \
 	      ] && echo 1)" "1"
