--- pcp-4.1.0/src/include/builddefs.in~	2018-06-15 03:48:02.000000000 +0200
+++ pcp-4.1.0/src/include/builddefs.in	2018-07-04 17:51:41.172261146 +0200
@@ -693,13 +693,7 @@ SETUP_PY_BUILD_OPTIONS = --include-dirs=
 SETUP_PY_BUILD_OPTIONS += --library-dirs=$(TOPDIR)/src/libpcp/src:$(TOPDIR)/src/libpcp_pmda/src:$(TOPDIR)/src/libpcp_gui/src:$(TOPDIR)/src/libpcp_import/src:$(TOPDIR)/src/libpcp_mmv/src
 SETUP_PY_INSTALL_OPTIONS = --skip-build
 SETUP_PY_INSTALL_OPTIONS += --root="$${DIST_ROOT:-/}"
-ifeq "$(PYTHON_PREFIX)" "/usr"
-ifeq "$(PACKAGE_DISTRIBUTION)" "debian"
-SETUP_PY_INSTALL_OPTIONS += --install-layout=deb
-endif
-else
 SETUP_PY_INSTALL_OPTIONS += --prefix=$(PYTHON_PREFIX)
-endif
 ifeq "$(ENABLE_PYTHON3)" "true"
 SETUP_PY_BUILD_OPTIONS += -DENABLE_PYTHON3
 endif
